### ISSUE TITLE
CLI arguments

### ISSUE THREAD

**Issue by Nevimzase** [microsoft/playwright-pytest#259]
Hi, I am begginer in Python-Pytest and learning CLI arguments. I have Python 3.12, Pytest 8.3.3 and Playwright 1.48 and none of these arguments work for me 

--headed: Run tests in headed mode (default: headless).
--browser: Run tests in a different browser chromium, firefox, or webkit. It can be specified multiple times (default: chromium).
--browser-channel [Browser channel](https://playwright.dev/python/docs/browsers) to be used.
--slowmo Slows down Playwright operations by the specified amount of milliseconds. Useful so that you can see what is going on (default: 0).
--device [Device](https://playwright.dev/python/docs/emulation) to be emulated.
--output Directory for artifacts produced by tests (default: test-results).
--tracing Whether to record a [trace](https://playwright.dev/python/docs/trace-viewer) for each test. on, off, or retain-on-failure (default: off).
--video Whether to record video for each test. on, off, or retain-on-failure (default: off).
--screenshot Whether to automatically capture a screenshot after each test. on, off, or only-on-failure (default: off).
--full-page-screenshot Whether to take a full page screenshot on failure. By default, only the viewport is captured. Requires --screenshot to be enabled (default: off).

https://playwright.dev/python/docs/test-runners

I don´t know why. It does not work for any tests - working, failing, simply do not react. For exemple --device does not emulate devices, --video and --screenshot do not create file with video and screenshot records, --output does not create wished directory, etc.

What is the reason and solution?

**Comment by mxschmitt** [comment]
Have you installed `pytest-playwright`? What is the output of `pip list`?

**Comment by Nevimzase** [comment]
Yes, pytest-playwright is installed pytest-playwright 0.5.2m. result of pip list is: Package Version
------- -------
pip     24.2

**Comment by Nevimzase** [comment]
> Have you installed `pytest-playwright`? What is the output of `pip list`? I recommend following this guide: https://playwright.dev/python/docs/intro

Yes, pytest-playwright is installed pytest-playwright 0.5.2m. result of pip list is: Package Version

pip 24.2

**Comment by mxschmitt** [comment]
Maybe you have it installed in different virtual environments? We have not heard of issues around this. e.g. try using [virtualenv](https://virtualenv.pypa.io/en/latest/user_guide.html) and do `pip uninstall pytest pytest-playwright` before.

**Comment by Nevimzase** [comment]
> Maybe you have it installed in different virtual environments? We have not heard of issues around this. e.g. try using [virtualenv](https://virtualenv.pypa.io/en/latest/user_guide.html) and do `pip uninstall pytest pytest-playwright` before.

I don´t think I installed it in different virtual environment. I have been taking an online course on Udemy and did same steps as the lector. Did not installed anything extra. Only the same packages. I don´t see any package for environments in my Interpreter. 

Still thank you for your help.

I will try to re-install pytest and pytest-playwright and see if it helps.

**Comment by mxschmitt** [comment]
Do you mind sharing more information on how you reproduce it?

Whats the output of `pytest`?

Whats the output of `pip list`?

Whats your OS?

How did you install Python?

Happy to help but its hard without these kind of information. Also what happens if you follow the `virtualenv` instructions from above and install pytest-playwright via that?

**Comment by Nevimzase** [comment]
> Do you mind sharing more information on how you reproduce it?
> 
> Whats the output of `pytest`?
> 
> Whats the output of `pip list`?
> 
> Whats your OS?
> 
> How did you install Python?
> 
> Happy to help but its hard without these kind of information. Also what happens if you follow the `virtualenv` instructions from above and install pytest-playwright via that?

Hi, before I was about to go your way I wanted to check once more everything so that I can say honestly I did everything I could to find out the reason by myself and just before I found something that I saw before but it did not occured to me this might be it:

"Note that CLI arguments are only applied to the default browser, context and page fixtures. If you create a browser, a context or a page with the API call like [browser.new_context()](https://playwright.dev/python/docs/api/class-browser#browser-new-context), the CLI arguments are not applied."  https://playwright.dev/python/docs/test-runners     

And in my test there is: 
def test_about_us_section_verbiage(playwright: Playwright):
    browser = playwright.chromium.launch()
    #browser = playwright.chromium.launch(headless=False)
    **page = browser.new_page()**

OR 

def test_TOP_09(playwright: Playwright) -> None:
    browser = playwright.chromium.launch()
    #browser = playwright.chromium.launch(headless=False)
    **context = browser.new_context()**

So I guess I found it, right? :D

And why it did not work with test_exemple.py? I don´t know. Maybe some typo? I tried it now and it works. It creates folder with video and screenshot and other CLI arguments work.

If you agree this is it, I thank you for your patience and time. If you say that it something else, I will send you the asked info.

**Comment by mxschmitt** [comment]
> So I guess I found it, right? 

yes

> Actually one more question about Playwright Inspector - when I try to create new test. It opens always with:

We can probably change the default to Pytest in Python: `playwright codegen --target=python-pytest`

**Comment by Nevimzase** [comment]
> We can probably change the default to Pytest in Python: `playwright codegen --target=python-pytest

I wanted to run a test with the advice you gave me to see if it works and i met another trouble (I feel ashamed with so many troubles). The test I created could not be run: 
(command): pytest test_teste.py
=============================================================== test session starts ===============================================================
platform win32 -- Python 3.12.7, pytest-8.3.3, pluggy-1.5.0
rootdir: (deleted)
configfile: pytest.ini
plugins: base-url-2.1.0, forked-1.6.0, playwright-0.5.2, reporter-0.5.3, reporter-html1-0.9.2, xdist-3.6.1
collected 0 items                                                                                                                                   

============================================================== no tests ran in 0.02s ============================================================== 
ERROR: file or directory not found: test_teste.py

I tried to run other tests -> same results.
So I tried only "pytest": =============================================================== test session starts ===============================================================
platform win32 -- Python 3.12.7, pytest-8.3.3, pluggy-1.5.0
rootdir: (deleted)
configfile: pytest.ini
plugins: base-url-2.1.0, forked-1.6.0, playwright-0.5.2, reporter-0.5.3, reporter-html1-0.9.2, xdist-3.6.1
collected 0 items                                                                                                                                   

============================================================== no tests ran in 0.02s ==============================================================  


So basically I can not run any tests with this command, HOWEVER I can run every single test manually with hitting "run" (file or single test). I wonder why? 
single tests have test_   prefix
test files have test_    prefix
I don´t remember to do some change that could hamper the commands. I also restarted pycharm
